mlrun/api/crud/model_monitoring/model_endpoint_store.py

```diff
@@ -0,0 +1,821 @@
+                                 earliest time.
+        :param metrics:          A list of metrics to return for the model endpoint. There are pre-defined metrics for
+                                 model endpoints such as predictions_per_second and latency_avg_5m but also custom
+                                 metrics defined by the user. Please note that these metrics are stored in the time
+                                 series DB and the results will be appeared under model_endpoint.spec.metrics.
+        :param feature_analysis: When True, the base feature statistics and current feature statistics will be added to
+                                 the output of the resulting object.
+
+        :return: A ModelEndpoint object.
+        """
+        pass
+
+    def list_model_endpoints(
+        self, model: str, function: str, labels: typing.List, top_level: bool
+    ):
+        """
+        Returns a list of endpoint unique ids, supports filtering by model, function,
+        labels or top level. By default, when no filters are applied, all available endpoint ids for the given project
+        will be listed.
+
+        :param model:           The name of the model to filter by.
+        :param function:        The name of the function to filter by.
+        :param labels:          A list of labels to filter by. Label filters work by either filtering a specific value
+                                of a label (i.e. list("key==value")) or by looking for the existence of a given
+                                key (i.e. "key").
+        :param top_level:       If True will return only routers and endpoint that are NOT children of any router.
+
+        :return: List of model endpoints unique ids.
+        """
+        pass
+
+
+class _ModelEndpointKVStore(_ModelEndpointStore):
+    """
+    Handles the DB operations when the DB target is from type KV. For the KV operations, we use an instance of V3IO
+    client and usually the KV table can be found under v3io:///users/pipelines/project-name/model-endpoints/endpoints/.
+    """
+
+    def __init__(self, access_key: str, project: str):
+        super().__init__(project=project)
+        # Initialize a V3IO client instance
+        self.access_key = access_key
+        self.client = mlrun.utils.v3io_clients.get_v3io_client(
+            endpoint=mlrun.mlconf.v3io_api, access_key=self.access_key
+        )
+        # Get the KV table path and container
+        self.path, self.container = self._get_path_and_container()
+
+    def write_model_endpoint(self, endpoint: mlrun.api.schemas.ModelEndpoint):
+        """
+        Create a new endpoint record in the KV table.
+
+        :param endpoint: ModelEndpoint object that will be written into the DB.
+        """
+
+        # Retrieving the relevant attributes from the model endpoint object
+        attributes = self.get_params(endpoint)
+        # Create or update the model endpoint record
```

Comment:
I think it would be helpful to rather explain why we are flattening the structure, where can we find the expected structure of the table in the database